src/v/raft/state_machine_manager.cc

```diff
@@ -276,6 +281,7 @@ ss::future<> state_machine_manager::do_apply_raft_snapshot(
           });
     }
     _next = model::next_offset(metadata.last_included_index);
+    background_apply_units.clear();
```

Comment:
what's the reason for clearing the units here compared to making `acquire_background_apply_mutexes` be RAII (e.g. `with_background_apply_mutexes(...`?

Reply:
Just wanted to make it explicit to the reader..

I wanted to do this.

```
ss::future<> state_machine_manager::do_apply_raft_snapshot(
  snapshot_metadata metadata,
  storage::snapshot_reader& reader,
  std::vector<ssx::semaphore_units> background_apply_units) {
```

over

```
ss::future<> state_machine_manager::do_apply_raft_snapshot(
  snapshot_metadata metadata,
  storage::snapshot_reader& reader,
  std::vector<ssx::semaphore_units>) {
```

that signifies these units are from bg_applicators but the former won't compile because `background_apply_units` remains unused, so I had to use it some how, so explicitly cleared the vector.